src/panels/lovelace/editor/add-entities-to-view.ts

```diff
@@ -2,69 +2,126 @@ import {
   fetchConfig,
   LovelaceConfig,
   saveConfig,
+  fetchDashboards,
 } from "../../../data/lovelace";
 import { HomeAssistant } from "../../../types";
 import { showSuggestCardDialog } from "./card-editor/show-suggest-card-dialog";
 import { showSelectViewDialog } from "./select-view/show-select-view-dialog";
+import { showAlertDialog } from "../../../dialogs/generic/show-dialog-box";
 
 export const addEntitiesToLovelaceView = async (
   element: HTMLElement,
   hass: HomeAssistant,
-  entities: string[],
-  lovelaceConfig?: LovelaceConfig,
-  saveConfigFunc?: (newConfig: LovelaceConfig) => void
+  entities: string[]
 ) => {
-  if ((hass!.panels.lovelace?.config as any)?.mode === "yaml") {
+  const dashboards = await fetchDashboards(hass);
+
+  const storageDashs = dashboards.filter(
+    (dashboard) => dashboard.mode === "storage"
+  );
+
+  if (
+    (hass!.panels.lovelace?.config as any)?.mode !== "storage" &&
+    !storageDashs.length
+  ) {
+    // no storage dashboards, just show the YAML config
     showSuggestCardDialog(element, {
       entities,
+      yaml: true,
     });
     return;
   }
-  if (!lovelaceConfig) {
+
+  let lovelaceConfig;
+  let urlPath: string | null = null;
+  if ((hass!.panels.lovelace?.config as any)?.mode === "storage") {
     try {
       lovelaceConfig = await fetchConfig(hass.connection, null, false);
-    } catch {
-      alert(
-        hass.localize(
-          "ui.panel.lovelace.editor.add_entities.generated_unsupported"
-        )
-      );
-      return;
+    } catch (e) {
+      // default dashboard is in generated mode
     }
   }
-  if (!lovelaceConfig.views.length) {
-    alert(
-      "You don't have any Lovelace views, first create a view in Lovelace."
-    );
-    return;
-  }
-  if (!saveConfigFunc) {
-    saveConfigFunc = async (newConfig: LovelaceConfig): Promise<void> => {
+
+  if (!lovelaceConfig && storageDashs.length) {
+    // find first dashoard not in generated mode
+    for (const storageDash of storageDashs) {
       try {
-        await saveConfig(hass!, null, newConfig);
-      } catch {
-        alert(
-          hass.localize("ui.panel.config.devices.add_entities.saving_failed")
+        // eslint-disable-next-line no-await-in-loop
+        lovelaceConfig = await fetchConfig(
+          hass.connection,
+          storageDash.url_path,
+          false
         );
+        urlPath = storageDash.url_path;
```

Comment:
Is this data not returned from `fetchDashboards`? If not, we should consider updating that ? This seems very wasteful.

Reply:
I don't understand what you mean; if we have config stored for the dashboard? 
Yeah, that would help if the backend told us, it would have to check for the files only ones.

Reply:
I meant the url path. Now we're in a for loop fetching all the dashboards ?

Reply:
We are in a loop to get the first not generated dashboard, as we can't add items to a generated storage dashboard